docs/source/mar_specification.rst

```diff
@@ -0,0 +1,137 @@
+* **README.md**: plain-language information on the model, how to use it, author information, etc. in Markdown format.
+* **license.txt**: software license attached to the model, can be left blank if no license needed.
+
+Archive Format
+==============
+
+The MAR directory and its contents can be compressed into a zip file to constitute a single file package. When unzipped into a directory this file will reproduce the above directory structure, and should itself also be named after the model it contains.
+
+The Torchscript file format is also just a zip file with a specific structure. When creating such an archive with `save_net_with_metadata` a MAR-compliant Torchscript file can be created by including the contents of `metadata.json` as the `meta_values` argument of the function, and other files included as `more_extra_files` entries. These will be stored in a `extras` directory in the zip file and can be retrieved with `load_net_with_metadata` or with any other library/tool that can read zip data. In this format the `model.*` files are obviously not needed by `README.md` and `license.txt` can be added as more extra files.
+
+metadata.json File
+==================
+
+This file contains the metadata information relating to the model, including what the shape and format of inputs and outputs are, what the meaning of the outputs are, what type of model is present, and other information. The JSON structure is a dictionary containing a defined set of keys with additional user-specified keys. The mandatory keys are as follows:
+
+* **version**: version of the stored model.
+* **monai_version**: version of MONAI the MAR was generated on, later versions expected to work.
+* **pytorch_version**: version of Pytorch the MAR was generated on, later versions expected to work.
+* **numpy_version**: version of Numpy the MAR was generated on, later versions expected to work.
+* **optional_packages_version**: dictionary relating optional package names to their versions, these packages are not needed but are recommended to be isntalled with this stated minimum version.
+* **task**: plain-language description of what the model is meant to do.
+* **description**: longer form plain-language description of what the model is, what it does, etc.
+* **authorship**: state author(s) of the model.
+* **copyright**: state model copyright.
+* **network_data_format**: defines the format, shape, and meaning of inputs and outputs to the model, contains keys "inputs" and "outputs" relating named inputs/outputs to their format specifiers (defined below).
+
+Tensor format specifiers are used to define input and output tensors and their meanings, and must be a dictionary containing at least these keys:
+
+* **type**: what sort of data the tensor represents: "image", "label", etc.
+* **format**: what format of information is stored: "magnitude", "hounsfield", "kspace", "segmentation", "multiclass", etc.
```

Comment:
What is `mulitclass`? I'm guessing this is a array of integers where each integer represents a class index. In this case, I would personally refer to this as a "label map"

Reply:
The idea with that was to allow multiple categories for each voxel, so multi-labeling is assigning a label to each voxel as a number, to have multi-class requires keeping N channels for N classes to handle voxels that would be referenced in multiple channels.